apps/molgenis-components/src/components/forms/baseInputs/BaseInputDecimal.vue

```diff
@@ -11,35 +10,45 @@
     :readonly="readonly"
     :required="required"
     @keypress="handleKeyValidity"
-    @input="emitIfValid"
+    @input="handleInputChanged"
   />
 </template>
 
-<script>
-import BaseInput from "./BaseInput.vue";
-import { isNumericKey, flipSign } from "../../utils";
+<script lang="ts">
 import constants from "../../constants";
+import { flipSign, isNumericKey } from "../../utils";
+import BaseInput from "./BaseInput.vue";
 
-const { CODE_MINUS } = constants;
+const { CODE_MINUS, CODE_PERIOD } = constants;
 
 export default {
   extends: BaseInput,
   methods: {
-    emitIfValid(event) {
-      const value = parseFloat(event.target.value);
+    handleInputChanged(event: any) {
+      const value = event.target?.value;
+      if (!value) {
+        this.$emit("update:modelValue", null);
+      } else {
+        this.emitIfValid(value);
+      }
+    },
+    emitIfValid(strValue: string) {
+      const periodValue = strValue.replace(",", ".");
```

Comment:
This replace has unwanted side effects. When pasting simple dutch values this works nicely, e.g. '1,32'.

But when pasting US data or larger data this is annoying. For example, 
* when I paste '100,000.32' I expect it to work but it doesn't. Instead I get '100'. 

And when I paste large Dutch numbers it also goes wrong. For example,
* when I paste '100.000,32' it suprisingly gives no validation error, but after save it is suddenly '100'

I would recommend to replace "," with "" instead of "." so the US behavior is consistent..
Future feature request might be localization.

Reply:
I thought we decided not to allow comma's... Also being able to copy paste anything is a known issue (I made an issue for that a while ago).
My suggestion is to only allow periods as decimal separator and besides that only digits as to have a working version for now, and also as we agreed on slack. Leaving further functionality for a future PR as we are getting way out of scope and clearly need more specification before we continue on such things.

Reply:
agreed